subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputIntegrationTest.groovy

```diff
@@ -474,7 +473,7 @@ class StaleOutputIntegrationTest extends AbstractIntegrationSpec {
 
             if (project.findProperty('assertRemoved')) {
                 ${taskName}.doFirst {
-                    assert !file('${outputDirPath}').exists()
+                    assert file('${outputDirPath}').directory
```

Comment:
❌ Same as above. We would need to create a file in outputDirPath and then check that this file has been removed.

Reply:
We do create stale files in there by calling `createStaleOutputs()`, no? I added a check that the directory is actually empty.

Reply:
Sounds good.